src/main/java/uk/gov/hmcts/reform/wataskmonitor/services/jobs/cleanup/CleanUpJobService.java

```diff
@@ -0,0 +1,193 @@
+        log.info("{} cleanUpJobConfig: {}", TASK_CLEAN_UP.name(), cleanUpJobConfig);
+
+        if (!cleanUpJobConfig.getAllowedEnvironment()
+            .contains(cleanUpJobConfig.getEnvironment().toLowerCase(Locale.ROOT))) {
+            log.info("{} is not enabled for this environment: {}",
+                TASK_CLEAN_UP.name(), cleanUpJobConfig.getEnvironment());
+            return false;
+        }
+
+        log.info("{} is enabled for this environment: {}", TASK_CLEAN_UP.name(), cleanUpJobConfig.getEnvironment());
+        return true;
+    }
+
+    public List<HistoricCamundaTask> retrieveProcesses() {
+        List<HistoricCamundaTask> historyTasks = camundaClient.getHistoryProcesses(
+            "0",
+            cleanUpJobConfig.getCamundaMaxResults(),
+            buildSearchQuery()
+        );
+        log.info("{} task(s) retrieved successfully from history", historyTasks.size());
+        return historyTasks;
+    }
+
+    public GenericJobReport deleteHistoricProcesses(List<HistoricCamundaTask> historicCamundaTasks,
+                                                    String serviceToken) {
+        if (historicCamundaTasks == null || historicCamundaTasks.isEmpty()) {
+            log.info("There was no task(s) to delete.");
+            return new GenericJobReport(0, emptyList());
+        } else {
+            List<GenericJobOutcome> outcomesList;
+
+            outcomesList = deleteHistoricProcessesFromCamunda(historicCamundaTasks, serviceToken);
+
+            return new GenericJobReport(historicCamundaTasks.size(), outcomesList);
+        }
+    }
+
+    public GenericJobReport deleteActiveProcesses(List<HistoricCamundaTask> historicCamundaTasks, String serviceToken) {
+
+        if (!isCleanActiveTaskAllowed()) {
+            return new GenericJobReport(0, emptyList());
+        }
+
+        if (historicCamundaTasks == null || historicCamundaTasks.isEmpty()) {
+            log.info("There was no active task(s) to delete.");
+            return new GenericJobReport(0, emptyList());
+        } else {
+            List<GenericJobOutcome> outcomesList;
+
+            outcomesList = deleteActiveProcessesFromCamunda(historicCamundaTasks, serviceToken);
+
+            return new GenericJobReport(historicCamundaTasks.size(), outcomesList);
+        }
+    }
+
+    private List<GenericJobOutcome> deleteHistoricProcessesFromCamunda(List<HistoricCamundaTask> historicCamundaTasks,
+                                                                       String serviceToken) {
+        List<GenericJobOutcome> outcomeList = new ArrayList<>();
+        boolean isSuccess = false;
+
+        List<String> historicProcessInstanceIds = historicCamundaTasks.stream()
+            .map(HistoricCamundaTask::getId).collect(Collectors.toList());
+
+
+        String body = prepareRequestBody(HISTORIC_PROCESS_DELETE_REQUEST, historicProcessInstanceIds);
+
+        try {
+            camundaClient.deleteHistoryProcesses(
```

Comment:
What happens if you have a partial success?  E.g. some processes are deleted and others are not?  Does Camunda return an error or success with errors list?

Reply:
we don't know what happens in camunda. I've added count check before and after but gave up due to make more calls. If some of them don't removed we will try one more at the next job run

Reply:
Ok.